sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentImpl.java

```diff
@@ -854,35 +858,43 @@ private CompletableFuture<PrepareResult> prepareResourceAsync(
                                                     // If no parent was provided, parent to the root resource.
                                                     log.excessive("Getting parent urn: t=%s, name=%s, custom=%s, remote=%s", type, name, custom, remote);
 
+                                                    // If no parent was provided, parent to the root resource.
                                                     var parentUrn = options.getParent().isPresent()
                                                             ? Internal.of(options.getParent().get().getUrn()).getValueOptional()
                                                             : this.rootResource.getRootResourceAsync(type);
                                                     return parentUrn.thenCompose(
                                                             (Optional<String> pUrn) -> {
                                                                 log.excessive("Got parent urn: t=%s, name=%s, custom=%s, remote=%s", type, name, custom, remote);
-                                                                var providerRef = custom
-                                                                        ? CompletableFutures.flipOptional(options.getProvider().map(p -> Internal.from(p).getRegistrationId()))
```

Comment:
The ternary operator was not very readable here, too many things going on at once IMO.

Use of `final` local var guarantees the same behaviour (set var exactly once).